### PR TITLE
Rollup emitter with its dependencies

### PR DESCRIPTION
**Pull request by pshao25**
Fix https://github.com/Azure/autorest.csharp/issues/2464

### CODE REVIEW

File: src/AutoRest.CSharp/AutoRest.CSharp.csproj [hunk 1]
```diff
@@ -41,6 +41,11 @@
     <Content Include="build\CodeGeneration.targets" PackagePath="build\Microsoft.Azure.AutoRest.CSharp.targets" />
     <Content Include="build\CodeGeneration.targets" PackagePath="buildMultiTargeting\Microsoft.Azure.AutoRest.CSharp.targets" />
   </ItemGroup>
+  
+  <ItemGroup>
+    <Content Include="../../node_modules/@azure-tools/cadl-csharp/**/*" PackagePath="tools\cadl-csharp" />
```
Comment (archerzz) [inline src/AutoRest.CSharp/AutoRest.CSharp.csproj:46]: I think this will cause typescript files to be built into the DLL.

Comment (pshao25) [reply src/AutoRest.CSharp/AutoRest.CSharp.csproj:46]: No, it puts js files into nuget package.

File: eng/pipelines/build.yml [hunk 2]
```diff
@@ -55,6 +55,10 @@ stages:
               npm ci
             displayName: "Install packages"
             workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - script: |
+              npm run build
+            displayName: "Build Emitter.Csharp"
```
Comment (lirenhe) [inline eng/pipelines/build.yml:60]: A small question, what is the official name? CSharp or csharp or Csharp

Comment (pshao25) [reply eng/pipelines/build.yml:60]: The folder name in source code is `Emitter.Csharp`. The official package name is `@azure-tools/cadl-csharp`

File: src/CADL.Extension/Emitter.Csharp/package.json [hunk 3]
```diff
@@ -17,34 +17,39 @@
         "cadl"
     ],
     "type": "module",
-    "main": "dist/src/index.js",
+    "main": "dist/bundle.js",
```
Comment (archerzz) [inline src/CADL.Extension/Emitter.Csharp/package.json:20]: will this have any impact on debugging emitter js codes?

**Pull request closed without merging**